Dockerfile

```diff
@@ -1,4 +1,4 @@
-FROM ruby:3-slim
+FROM ruby:3.0.3
```

Comment:
Thank you!  We can also bring the Ruby version up to 3.1 in both places. Why don't we do `3.1.2-slim` here and update the asdf `.tools-versions` file and the Gemfile to `3.1.2`. We don't need the full ruby install for the dev container for anything yet, that I'm aware of.